pl_bolts/utils/semi_supervised.py

```diff
@@ -24,14 +26,14 @@ class Identity(torch.nn.Module):
         model.fc = Identity()
 
     """
-    def __init__(self):
+    def __init__(self) -> None:
         super(Identity, self).__init__()
 
-    def forward(self, x):
+    def forward(self, x: Tensor) -> Tensor:
         return x
 
 
-def balance_classes(X: np.ndarray, Y: list, batch_size: int):
+def balance_classes(X: np.ndarray, Y: list, batch_size: int) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
May we be more specific here?
```suggestion
def balance_classes(X: np.ndarray, Y: List[int], batch_size: int) -> Tuple[np.ndarray, np.ndarray]:
```

Reply:
My bad, I think I missed this because it was already annotated.😄

Reply:
Should I change it to `Sequence[int]` since it can be a `list` or a `tuple`?

Reply:
@hassiahk On second thought, I don't think the annotated types are not correct. In the tests, we use the function as the following:
```python
tests/utils/test_semi_supervised.py:9:def test_balance_classes():
tests/utils/test_semi_supervised.py-10-    X = torch.rand(100, 3, 32, 32)
tests/utils/test_semi_supervised.py-11-    c1 = torch.zeros(20, 1)
tests/utils/test_semi_supervised.py-12-    c2 = torch.zeros(20, 1) + 1
tests/utils/test_semi_supervised.py-13-    c3 = torch.zeros(60, 1) + 2
tests/utils/test_semi_supervised.py-14-
tests/utils/test_semi_supervised.py-15-    y = torch.cat([c1, c2, c3], dim=0).int().cpu().numpy().flatten().tolist()
tests/utils/test_semi_supervised.py:16:    (balanced_X, balanced_Y) = balance_classes(X, y, batch_size=10)
```
which suggests we should annotate them differently from the current ones.

Reply:
Annotated `X` as `Union[Tensor, np.ndarray]` and `Y` as `Union[Tensor, np.ndarray, Sequence[int]]`.